modules/gdscript/gdscript_parser.cpp

```diff
@@ -2783,12 +2783,11 @@ void GDScriptParser::_parse_block(BlockNode *p_block, bool p_static) {
 
 				tokenizer->advance();
 				int var_line = tokenizer->get_token_line();
+				StringName n = tokenizer->get_token_literal();
```

Comment:
It's not safe to call `get_token_literal()` before checking if the current token really is a literal. If it's not, it'll throw a cryptic error message to the user from the tokenizer.

Reply:
Was the prior state with the call to `get_token_literal` after that point unsafe or was the error at that point intended?

Reply:
Before this change it was done after the `if` block, which  will `return` if it's not a literal. It would only reach the `get_token_literal()` call if the `is_token_literal()` were true, so it was safe.